core/models/access_token.py

```diff
@@ -0,0 +1,24 @@
+from django.db import models
+from django.utils import timezone
+
+from django_cyverse_auth.models import Token
+
+
+class AccessToken(models.Model):
+    """
+    Extend the django_cyverse_auth Token to add a name
+    """
+    token = models.OneToOneField(Token, on_delete=models.CASCADE)
+    name = models.CharField(max_length=128, null=False, blank=False)
+
+    class Meta:
+        db_table = "access_token"
+        app_label = "core"
+
+def create_access_token(user, token_name=None, token_expire=None, remote_ip=None, issuer=None):
+    if AccessToken.objects.filter(name=token_name, token__user_id=user.id):
+        return None
+    token = Token(user=user, issuer=issuer)
+    token.save()
+    access_token, created = AccessToken.objects.update_or_create(token=token, name=token_name)
```

Comment:
This may have been a recent change, but you don't want `update_or_create`. You just want create, a new token should always be returned, since a new token is always created.